homeassistant/components/usb/__init__.py

```diff
@@ -265,3 +271,14 @@ async def websocket_usb_scan(
     if not usb_discovery.observer_active:
         await usb_discovery.async_request_scan_serial()
     connection.send_result(msg["id"])
+
+
+@callback
+def async_is_plugged_in(hass: HomeAssistant, matcher: Mapping) -> bool:
+    """Return True is a USB device is present."""
+    usb_discovery: USBDiscovery = hass.data[DOMAIN]
+    for device_tuple in usb_discovery.seen:
+        device = USBDevice(*device_tuple)
+        if _is_matching(device, matcher):
+            return True
+    return False
```

Comment:
This method will return `True` even if a device has been unplugged again.

Reply:
We need to solve that in a future PR as it relies tying into the OS